src/haz3lweb/view/Code.re

```diff
@@ -5,30 +5,86 @@ open Util;
 open Util.Web;
 
 let of_delim' =
-  Core.Memo.general(
-    ~cache_size_bound=100000,
-    ((sort, is_consistent, is_complete, label, i)) => {
-      let cls =
-        switch (label) {
-        | [_] when !is_consistent => "mono-inconsistent"
-        | [s] when Form.is_string(s) => "mono-string-lit"
-        | [_] => "mono"
-        | _ when !is_consistent => "delim-inconsistent"
-        | _ when !is_complete => "delim-incomplete"
-        | _ => "delim"
-        };
-      [
-        span(
+    ((sort, is_consistent, is_complete, label, i, inject, livelit_state)) => {
```

Comment:
Not sure what to do about this. `inject` is a function therefore unhashable so this function doesn't get memoized. There's probably a smarter way to do the event handling I'm not thinking of. I'd rather not have to do something that reflects over the dom and captures the input elements.

Reply:
I did end up refactoring this a bit in the livelit params branch https://github.com/hazelgrove/hazel/commit/2b8b9a2e2adef8575237f633b40cc275d99bef07#diff-98923b69b71582fa885895acfb8efef0a607453ecf764a5cedd4021db6521d3b. I was trying to separate out those changes but I can see if that parts easy to reason in and if it would simplify.